### ISSUE TITLE
nginx: certbot seems to detect commented entries in server_name

### ISSUE THREAD

**Issue by robsch** [certbot/certbot#7090]
Found this with version 0.31.0, but I guess it is still the same with the current version.

If there is a nginx configuration file that contains:

    server {
        server_name  somedomain.com
                  # anotherdomain.com
        ;
    }

And there is another server configuration with: 

    server {
        server_name  anotherdomain.com;
    }
 
It could happen that the second server configuration does not get detected (I guess it is a question about the sequence how files get scanned). So certbot should ignore commented server_name entries.

**Comment by adferrand** [comment]
Hello!

Could you provide the letsencrypt log on an example of that Certbot execution?

**Comment by robsch** [comment]
Unfortunatly not. But I have a server configuration with server_name entry that is commented out. And if I call `sudo certbot --nginx` I get this entry listed. If I'm not wrong it should be not so hard to filter those lines with a regular expression, right?

**Comment by adferrand** [comment]
Well, you found a valid corner case, that I could reproduce also.

So, most of the time, commented lines are effectively ignored. For instance, following situation is correctly handled:
```
server_name foo.example.com;
# server_name bar.example.com;
```

However, in the pathological situation where a directive is on three lines, with a comment in the middle:
```
server_name foo.example.com
# bar.example.com
;
```

Then the parser is tricked, and interpret the lines as if it was:
```
server_name foo.example.com bar.example.com;
```

Thanks for your report!

cc @ohemorange

**Comment by robsch** [comment]
Yes, just a minor issue. But I thought I better report it - for the case you don't what to do next ;-)
